### ISSUE TITLE
StrictMode policy violation

### ISSUE THREAD

**Issue by frankjoshua** [adaptyteam/AdaptySDK-Android#26]
Looks like in PreferenceManager you are using commit() instead of apply(). Should be simple to fix. This is important though because on some phones saving even a simple preference can take many seconds and cause ANRs.

```
StrictMode policy violation; ~duration=11 ms: android.os.strictmode.DiskWriteViolation
                                                                                                    	at android.os.StrictMode$AndroidBlockGuardPolicy.onWriteToDisk(StrictMode.java:1622)
                                                                                                    	at libcore.io.BlockGuardOs.rename(BlockGuardOs.java:372)
                                                                                                    	at libcore.io.ForwardingOs.rename(ForwardingOs.java:698)
                                                                                                    	at android.app.ActivityThread$AndroidOs.rename(ActivityThread.java:8173)
                                                                                                    	at java.io.UnixFileSystem.rename(UnixFileSystem.java:400)
                                                                                                    	at java.io.File.renameTo(File.java:1410)
                                                                                                    	at android.app.SharedPreferencesImpl.writeToFile(SharedPreferencesImpl.java:786)
                                                                                                    	at android.app.SharedPreferencesImpl.-$$Nest$mwriteToFile(Unknown Source:0)
                                                                                                    	at android.app.SharedPreferencesImpl$1.run(SharedPreferencesImpl.java:680)
                                                                                                    	at android.app.SharedPreferencesImpl.enqueueDiskWrite(SharedPreferencesImpl.java:699)
                                                                                                    	at android.app.SharedPreferencesImpl.-$$Nest$menqueueDiskWrite(Unknown Source:0)
                                                                                                    	at android.app.SharedPreferencesImpl$EditorImpl.commit(SharedPreferencesImpl.java:612)
                                                                                                    	at com.adapty.internal.data.cache.PreferenceManager.saveLong(PreferenceManager.kt:54)
                                                                                                    	at com.adapty.internal.data.cache.CacheRepository.saveLastAppOpenedTime(CacheRepository.kt:211)
                                                                                                    	at com.adapty.internal.utils.LifecycleAwareRequestRunner.handleAppOpenedEvent$lambda$0(LifecycleAwareRequestRunner.kt:73)
                                                                                                    	at com.adapty.internal.utils.LifecycleAwareRequestRunner.$r8$lambda$AZACOowsydQgEC-9ewPayGLIp3o(Unknown Source:0)
                                                                                                    	at com.adapty.internal.utils.LifecycleAwareRequestRunner$$ExternalSyntheticLambda0.onResult(Unknown Source:4)
                                                                                                    	at com.adapty.internal.data.cloud.AnalyticsEventRecorder$trackEvent$1$2$2.invokeSuspend(AnalyticsEventRecorder.kt:59)
                                                                                                    	at com.adapty.internal.data.cloud.AnalyticsEventRecorder$trackEvent$1$2$2.invoke(Unknown Source:8)
                                                                                                    	at com.adapty.internal.data.cloud.AnalyticsEventRecorder$trackEvent$1$2$2.invoke(Unknown Source:4)
                                                                                                    	at kotlinx.coroutines.flow.FlowKt__TransformKt$onEach$$inlined$unsafeTransform$1$2.emit(Emitters.kt:223)
                                                                                                    	at kotlinx.coroutines.flow.FlowKt__ErrorsKt$catchImpl$2.emit(Errors.kt:158)
                                                                                                    	at kotlinx.coroutines.flow.FlowKt__ChannelsKt.emitAllImpl$FlowKt__ChannelsKt(Channels.kt:37)
                                                                                                    	at kotlinx.coroutines.flow.FlowKt__ChannelsKt.access$emitAllImpl$FlowKt__ChannelsKt(Channels.kt:1)
                                                                                                    	at kotlinx.coroutines.flow.FlowKt__ChannelsKt$emitAllImpl$1.invokeSuspend(Unknown Source:14)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:205)
                                                                                                    	at android.os.Looper.loop(Looper.java:294)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8248)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

**Comment by vladd-g** [comment]
Thank you, @frankjoshua, we're taking a look

**Comment by vladd-g** [comment]
fixed in v2.11.0
